### PR TITLE
Microservice DB implementation

### PR DESCRIPTION
**Pull request by fjavierponce**
This PR contains the database implementation for the microservice module. 
- Repository implementation
- Stored procedures integrated with docker compose DB
- Rabbitmq and Postgres on docker compose

### CODE REVIEW

File: skt-microservice/src/main/java/com/nearsoft/skt/repository/impl/ProductRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    @Override
+    @Transactional
+    public List<Product> getAllProducts() {
+        StoredProcedureQuery storedProcedureQuery = entityManager.createStoredProcedureQuery("get_products");
+        storedProcedureQuery.registerStoredProcedureParameter(1, void.class, ParameterMode.REF_CURSOR);
+        storedProcedureQuery.execute();
+        List<Object[]>  productsFromStoreProcedure = storedProcedureQuery.getResultList();
+        List<Product> products = new ArrayList<>();
```
Comment (ragore09) [inline skt-microservice/src/main/java/com/nearsoft/skt/repository/impl/ProductRepositoryImpl.java]: It seems you're not using `products` anywhere in the method

File: skt-microservice/src/main/java/com/nearsoft/skt/repository/impl/ProductRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+@Repository
+class ProductRepositoryImpl implements ProductRepository {
+
+    @PersistenceContext
+    EntityManager entityManager;
+
+    @Override
+    @Transactional
+    public void createProduct(com.nearsoft.skt.model.Product product) {
+        entityManager.createNativeQuery("call createProduct(:inParam1, :inParam2); END;")
```
Comment (wombstrat) [inline skt-microservice/src/main/java/com/nearsoft/skt/repository/impl/ProductRepositoryImpl.java]: I really like the idea of using named parameteres, but could you use a more descriptive name param?, `InParam1` is not much descriptive.

File: skt-microservice/src/main/resources/application.yml [hunk 3]
```diff
@@ -2,5 +2,10 @@ spring:
   profiles:
     include:
       - common
+  datasource:
+    driverClassName: org.postgresql.Driver
+    url: jdbc:postgresql://localhost:5432/skt_products
```
Comment (wombstrat) [inline skt-microservice/src/main/resources/application.yml]: Nitpick: use other yml properties for host url and port.

File: skt-microservice/src/main/java/com/nearsoft/skt/repository/impl/ProductRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Repository
+class ProductRepositoryImpl implements ProductRepository {
+
+    @PersistenceContext
+    EntityManager entityManager;
```
Comment (wombstrat) [inline skt-microservice/src/main/java/com/nearsoft/skt/repository/impl/ProductRepositoryImpl.java]: Might be better use contructor based injection.

File: skt-microservice/src/main/java/com/nearsoft/skt/handler/MessageHandler.java [hunk 5]
```diff
@@ -38,7 +39,6 @@ public void handleCreateProductMessage(Product product) {
     @RabbitListener(queues = "#{sktCommonProperties.getQueues().getConsumerQueue()}")
     public List<Product> handleGetProductsMessage(String message) {
         logger.info("Received request '" + message + "' to return list of products. Returning list of products." );
```
Comment (wombstrat) [inline skt-microservice/src/main/java/com/nearsoft/skt/handler/MessageHandler.java]: Nitpick: Use logger interpolation feature `logger.info("Message: {}", message)`

File: skt-microservice/src/main/java/com/nearsoft/skt/repository/impl/ProductRepositoryImpl.java (hunk 1 above)
Comment (fjavierponce) [reply skt-microservice/src/main/java/com/nearsoft/skt/repository/impl/ProductRepositoryImpl.java]: my bad, I'm changing it.

File: skt-microservice/src/main/java/com/nearsoft/skt/repository/impl/ProductRepositoryImpl.java (hunk 2 above)
Comment (fjavierponce) [reply skt-microservice/src/main/java/com/nearsoft/skt/repository/impl/ProductRepositoryImpl.java]: agree.

File: skt-microservice/src/main/java/com/nearsoft/skt/repository/impl/ProductRepositoryImpl.java (hunk 4 above)
Comment (fjavierponce) [reply skt-microservice/src/main/java/com/nearsoft/skt/repository/impl/ProductRepositoryImpl.java]: Agree, I am changing it.

File: skt-microservice/src/main/java/com/nearsoft/skt/handler/MessageHandler.java (hunk 5 above)
Comment (fjavierponce) [reply skt-microservice/src/main/java/com/nearsoft/skt/handler/MessageHandler.java]: Agree, I know is better. My bad, thanks.

**Pull request merged**